rules/DeadCode/PhpDoc/DeadVarTagValueNodeAnalyzer.php

```diff
@@ -35,6 +36,12 @@ public function isDead(VarTagValueNode $varTagValueNode, Property $property): bo
         $propertyType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($property->type);
         $docType = $this->staticTypeMapper->mapPHPStanPhpDocTypeNodeToPHPStanType($varTagValueNode->type, $property);
 
+        // NonExistingObjectType may refer to @template tag defined in class
+        if ($docType instanceof NonExistingObjectType && ! str_contains($docType->getClassName(), '\\')) {
+            dump($docType);
+            return false;
+        }
```

Comment:
@dragosprotung  this can be tweak solution, but it will cause invalid skipped real non existing object type, see fixture https://github.com/rectorphp/rector-src/pull/6396/files#diff-d9d0d07e6a9125b667f68974544d4c6567c6b31da986c9644d725c188257f378

```diff
There was 1 failure:

1) Rector\Tests\DeadCode\Rector\Property\RemoveUselessVarTagRector\RemoveUselessVarTagRectorTest::test with data set #3 ('/home/runner/work/rector-src/...hp.inc')
Failed on fixture file "non_existing_object.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 final class NonExistingObject
 {
+    /** @var SomeNonExistingObject */
     private Properties|null $properties;
```

Reply:
The patch seems needs to be in `ObjectTypeSpecifier` to locate `@template` if exists before marking as `NonExistingObjectType`

https://github.com/rectorphp/rector-src/blob/d4152608a5983e813adbd0a9e45d213c722c1347/rules/TypeDeclaration/PHPStan/ObjectTypeSpecifier.php#L58-L70

Reply:
Yes, you might be right. Unfortunatly I do not have enought knowlage of PHPStan to find out if the type is a template or not.

Reply:
`@template` tag should be detected, unknown type can be just typo missing FQCN on class name, which should be removed as pointed to invalid class target.

Reply:
Fixed by get `@template` from `ClassReflection`  https://github.com/rectorphp/rector-src/pull/6396/commits/02b566a121558c6d2756f128cedf24e2065f3f88

current limitation is currently only detect single `@template` that can be improved in separate PR when possible :)

Reply:
It seems cause invalid rectify https://github.com/rectorphp/rector-src/pull/6396/commits/366194ba4a240545eb59d52217b9a785ecb18f41 , I will check more :)

Reply:
should be fixed now :)